tests/e2e/vmlistener_test.go

```diff
@@ -346,6 +352,159 @@ func (s *VMListenerTestParams) TestFailoverAcrossVmOnVsan(c *C) {
 	misc.LogTestEnd(c.TestName())
 }
 
+// TestVolumeAttachedForHostdRestart - verify an in use volume
+// remains attached when hostd is restarted.
+// 1. Restart hostd
+// 2. Verify volume stays as attached
+// 3. Kill the vm
+// 4. Status of the volume should be detached.
+func (s *VMListenerTestParams) TestVolumeAttachedForHostdRestart(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	// Create the volume
+	out, err := dockercli.CreateVolume(s.vm1, s.volumeName)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Attach the volume
+	out, err = dockercli.AttachVolume(s.vm1, s.volumeName, s.containerName)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// 1. Restart hostd
+	out, err = ssh.InvokeCommand(s.esx, restartHostd)
+	c.Assert(err, IsNil, Commentf(out))
+
+	// Give hostd time to start up
+	misc.SleepForSec(30)
```

Comment:
how about polling on hostd status ... 30 seconds is going to big I guess. This PR itself is adding 90 seconds of static wait time.